src/request/sign-polygon-transaction/SignPolygonTransactionApi.js

```diff
@@ -0,0 +1,117 @@
+                    request.approval.tokenNonce,
+                    true,
+                    'approval.tokenNonce',
+                ),
+            };
+        }
+
+        return parsedRequest;
+    }
+
+    /**
+     *
+     * @param {KeyguardRequest.PolygonTransactionInfo} request
+     * @param {string[]} allowedMethods
+     * @returns {PolygonTransferDescription | PolygonTransferWithApprovalDescription | PolygonRefundDescription}
+     */
+    parseOpenGsnForwardRequest(request, allowedMethods) {
+        request.request = this.parseOpenGsnForwardRequestRoot(request.request);
+
+        /** @type {PolygonTransferDescription | PolygonTransferWithApprovalDescription | PolygonRefundDescription} */
+        let description;
+
+        if (request.request.to === CONFIG.USDC_TRANSFER_CONTRACT_ADDRESS) {
+            const usdcTransferContract = new ethers.Contract(
+                CONFIG.USDC_TRANSFER_CONTRACT_ADDRESS,
+                PolygonContractABIs.USDC_TRANSFER_CONTRACT_ABI,
+            );
+
+            /** @type {PolygonTransferDescription | PolygonTransferWithApprovalDescription} */
+            description = (usdcTransferContract.interface.parseTransaction({
+                data: request.request.data,
+                value: request.request.value,
+            }));
+        } else if (request.request.to === CONFIG.USDC_HTLC_CONTRACT_ADDRESS) {
+            const usdcHtlcContract = new ethers.Contract(
+                CONFIG.USDC_HTLC_CONTRACT_ADDRESS,
+                PolygonContractABIs.USDC_HTLC_CONTRACT_ABI,
+            );
+
+            /** @type {PolygonRefundDescription} */
+            description = (usdcHtlcContract.interface.parseTransaction({
+                data: request.request.data,
+                value: request.request.value,
+            }));
+        } else {
+            throw new Errors.InvalidRequestError('request.to address is not allowed');
+        }
+
+        if (!allowedMethods.includes(description.name)) {
+            throw new Errors.InvalidRequestError('Requested Polygon contract method is invalid');
+        }
+
+        if (description.name === 'transfer' || description.name === 'transferWithApproval') {
+            if (description.args.token !== CONFIG.USDC_CONTRACT_ADDRESS) {
+                throw new Errors.InvalidRequestError('Invalid USDC token contract in request data');
+            }
+        }
+
+        // Check that amount exists when method is 'refund'
+        if (description.name === 'refund') {
+            if (!request.amount) {
+                throw new Errors.InvalidRequestError('`amount` required for refund method');
+            }
+        }
+
+        // Check that approval object exists when method is 'transferWithApproval'
+        if (description.name === 'transferWithApproval') {
+            if (!request.approval) {
+                throw new Errors.InvalidRequestError('`approval` object required for transferWithApproval method');
+            }
+        }
```

Comment:
```suggestion
        // Check that approval object exists when method is 'transferWithApproval', and unset for other methods.
        if ((description.name === 'transferWithApproval') !== !!request.approval) {
            throw new Errors.InvalidRequestError('`approval` object required for transferWithApproval method and ignored for other methods');
        }
```
`approval` is required only for `transferWithApproval` and ignored otherwise.